### PR TITLE
complete sumologic

### PR DESCRIPTION
**Pull request by mbliakharskyi**
## Description

Please include a summary of the changes and issues addressed in this PR. Also, explain the motivation and context of each update. List any dependencies or prerequisites needed for implementing this change.

## Checklist:

Before you create this PR, confirm all the requirements listed below by checking the checkboxes like this (`[x]`).

- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added tests to cover the new feature or fixes.

### CODE REVIEW

File: apps/sumologic/src/connectors/users.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+export type DeleteUsersParams = {
+  userId: string;
+  accessId: string;
+  accessKey: string;
+};
+
+export const getUsers = async ({ accessId, accessKey, afterToken }: GetUsersParams) => {
+  const encodedKey = Buffer.from(`${accessId}:${accessKey}`).toString('base64');
+
+  const url = new URL(`${env.SUMOLOGIC_API_BASE_URL}v1/users`);
```
[the remark this answers is not in the document]
Comment (mbliakharskyi) [reply apps/sumologic/src/connectors/users.ts]: done

File: apps/sumologic/src/app/install/page.tsx [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+    AU: 'https://api.au.sumologic.com/api/',
+    CA: 'https://api.ca.sumologic.com/api/',
+    DE: 'https://api.de.sumologic.com/api/',
+    EU: 'https://api.eu.sumologic.com/api/',
+    FED: 'https://api.fed.sumologic.com/api/',
+    IN: 'https://api.in.sumologic.com/api/',
+    JP: 'https://api.jp.sumologic.com/api/',
+    US1: 'https://api.us1.sumologic.com/api/',
+    US2: 'https://api.us2.sumologic.com/api/',
+  };
```
Comment (elbalexandre) [inline apps/sumologic/src/app/install/page.tsx:99]: Let's exports this from `action.tsx`, so we can share it in the zod schema + component
```suggestion
export  const sourceRegions = [
    'au',
    'ca',
    'de',
    'eu',
    'fed',
    'in',
    'jp',
    'us1',
    'us2'
]
```

File: apps/sumologic/src/app/install/page.tsx [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+                    name="sourceRegion"
+                    onChange={(e) => {
+                      setSelectedDomain(e.target.value);
+                    }}
+                    value={selectedDomain}>
+                    {Object.entries(domainOptions).map(([key, value]) => (
+                      <option key={value} value={key}>
+                        {key}
+                      </option>
+                    ))}
```
Comment (elbalexandre) [inline apps/sumologic/src/app/install/page.tsx:191]:
```suggestion
                    {sourceRegions.map((sourceRegion) => (
                      <option key={sourceRegion} value={sourceRegion}>
                        {sourceRegion.toUpperCase()}
                      </option>
                    ))}
```

File: apps/sumologic/src/app/install/page.tsx [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+                </div>
+                <div role="group">
+                  <label htmlFor="sourceRegion">Deployment Region</label>
+                  <select
+                    id="sourceRegion"
+                    name="sourceRegion"
+                    onChange={(e) => {
+                      setSelectedDomain(e.target.value);
+                    }}
+                    value={selectedDomain}>
```
Comment (elbalexandre) [inline apps/sumologic/src/app/install/page.tsx:186]:
```suggestion
                    >
```
I don't think we need to store the select value in a state.

File: apps/sumologic/src/app/install/action.tsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+import { z } from 'zod';
+import { SumologicError } from '@/connectors/commons/error';
+import { env } from '@/env';
+import { registerOrganisation } from './service';
+
+const formSchema = z.object({
+  organisationId: z.string().uuid(),
+  accessId: z.string().min(1),
+  accessKey: z.string().min(1),
+  sourceRegion: z.enum(['AU', 'CA', 'DE', 'EU', 'FED', 'IN', 'JP', 'US1', 'US2']),
```
Comment (elbalexandre) [inline apps/sumologic/src/app/install/action.tsx:12]: Using suggested `sourceRegions`.

```suggestion
  sourceRegion: z.enum(sourceRegions),
```

File: apps/sumologic/src/connectors/users.ts [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+  userId: string;
+  accessId: string;
+  accessKey: string;
+  sourceRegion: string;
+};
+
+export const getUsers = async ({ accessId, accessKey, afterToken, sourceRegion }: GetUsersParams) => {
+  const encodedKey = Buffer.from(`${accessId}:${accessKey}`).toString('base64');
+
+  const url = new URL(`https://api.${sourceRegion.toLowerCase()}.sumologic.com/api/v1/users`);
```
Comment (elbalexandre) [inline apps/sumologic/src/connectors/users.ts:37]:
```suggestion
  const url = new URL(`https://api.${sourceRegion}.sumologic.com/api/v1/users`);
```
We don't need to lower the case if we make sure sourceRegion is always in lowercase.

File: apps/sumologic/src/connectors/users.ts [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+
+  return {
+    validUsers,
+    invalidUsers,
+    nextPage: next ? next : null,
+  };
+};
+
+export const deleteUsers = async ({ userId, accessId, accessKey, sourceRegion }: DeleteUsersParams) => {
+  const url = new URL(`https://api.${sourceRegion.toLowerCase()}.sumologic.com/api/v1/users/${userId}`);
```
Comment (elbalexandre) [inline apps/sumologic/src/connectors/users.ts:76]:
```suggestion
  const url = new URL(`https://api.${sourceRegion}.sumologic.com/api/v1/users/${userId}`);
```

File: apps/sumologic/src/app/install/page.tsx (hunk 4 above)
Comment (mbliakharskyi) [reply apps/sumologic/src/app/install/page.tsx:186]: Okay.

**Pull request merged**